### PR TITLE
Removes all wrapped errors in clickhouse.go

### PR DESCRIPTION
**Pull request by epps**
# Description

Removes all wrapped errors in clickhouse.go.

### Select type(s) of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

# Checklist:

- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] I have fixed any merge conflicts

### CODE REVIEW

**Comment by sdreyer** [comment]
Got a conflict here

File: provider/online.go [hunk 1]
```diff
@@ -125,15 +94,19 @@ func (store *localOnlineStore) AsOnlineStore() (OnlineStore, error) {
...
 	}
 	return table, nil
 }
 
 func (store *localOnlineStore) CreateTable(feature, variant string, valueType ValueType) (OnlineStoreTable, error) {
 	key := tableKey{feature, variant}
 	if _, has := store.tables[key]; has {
-		return nil, &TableAlreadyExists{feature, variant}
+		wrapped := fferr.NewDatasetAlreadyExistsError(feature, variant, nil)
+		wrapped.AddDetail("provider", store.ProviderType.String())
```
Comment (anthonylasso) [inline provider/online.go:108]: it's not possible for store.ProviderType to be nil right?

Comment (epps) [reply provider/online.go]: It's possible inasmuch as we don't validate this, but it shouldn't happen.

**Comment by epps** [comment]
> Got a conflict here

Rebased on `other/errorhandling`

File: provider/clickhouse.go [hunk 2]
```diff
@@ -269,8 +274,12 @@ func (table *clickhouseOfflineTable) Write(rec ResourceRecord) error {
 		return err
 	}
 	insertQuery := table.query.writeInserts(tb)
-	_, err := table.db.Exec(insertQuery, rec.Entity, rec.Value, rec.TS)
-	return err
+	if _, err := table.db.Exec(insertQuery, rec.Entity, rec.Value, rec.TS); err != nil {
+		wrapped := fferr.NewResourceExecutionError(pt.ClickHouseOffline.String(), rec.Entity, "", fferr.ENTITY, err)
```
Comment (sdreyer) [inline provider/clickhouse.go:278]: The name variant should be the name variant of the table right?

File: provider/clickhouse.go [hunk 3]
```diff
@@ -835,11 +900,11 @@ func (store *clickHouseOfflineStore) UpdateMaterialization(id ResourceID) (Mater
 
 	rows, err := store.db.Query(getMatQry, tableName)
 	if err != nil {
-		return nil, err
+		return nil, fferr.NewResourceExecutionError(pt.ClickHouseOffline.String(), id.Name, id.Variant, fferr.ResourceType(id.Type.String()), err)
 	}
 	defer rows.Close()
 	if !rows.Next() {
-		return nil, &MaterializationNotFound{matID}
+		return nil, fferr.NewDatasetNotFoundError(id.Name, id.Variant, nil)
```
Comment (sdreyer) [inline provider/clickhouse.go:907]: Might want to specify that the table is empty

File: provider/clickhouse.go [hunk 4]
```diff
@@ -858,11 +923,13 @@ func (store *clickHouseOfflineStore) DeleteMaterialization(id MaterializationID)
 	if exists, err := store.materializationExists(id); err != nil {
 		return err
 	} else if !exists {
-		return &MaterializationNotFound{id}
+		return fferr.NewDatasetNotFoundError(string(id), "", nil)
```
Comment (sdreyer) [inline provider/clickhouse.go:926]: we need to fix this materialization naming

Comment (sdreyer) [reply provider/clickhouse.go:926]: not now but it makes things a pain

**Review by sdreyer — APPROVED** [review]

File: provider/clickhouse.go (hunk 2 above)
Comment (epps) [reply provider/clickhouse.go:278]: The feature variant is serialized into the table name at this point, so those aren't accessible.

**Pull request merged**